### DISCUSSION TITLE
Mutex / Queue - Execution of Job on a single node

### DISCUSSION THREAD

**Discussion by khmarbaise** [jobrunr/jobrunr#133] [Q&A] [answered]
Hi to all,

first some context. I would like to start at least two nodes on which of each a job can run. But I would like to make sure that the job is only executed on one of the nodes. 

Based on my current understanding the Mutex would be the way to go ? But how could that being done? Or is a mechanism already there to handle such things?

**Comment by rdehuyss** [comment, marked as answer]
Hi @khmarbaise , 

Even with multiple nodes running, if a job is only enqueued once, it will only run once - if there is 1 node or if there are 10 nodes.

If the same job is enqueued more than once (e.g. on each node), you can pass a UUID when you enqueue the job. If that UUID is the same, the job will only be enqueued once.

This comes in handy when you have for example different servers/instances all listening to the same service bus messages - your service bus message than typically contains a correlation UUID that you can use to make sure the job is only enqueued once.

Kind regards,
Ronald
